### PR TITLE
chirp: 0.4.0-unstable-2024-05-24 -> 0.4.0-unstable-2024-07-05, python311Packages.suds: init at 1.1.2

### PR DESCRIPTION
**Pull request by wrmilling**
## Description of changes

There are three main things in this PR: 

- new mandatory chirp dependency added, `python311Packages.suds: init at 1.1.2`, as mentioned in #317593.
- chirp package updated to 0.4.0-unstable-2024-07-05 and python 3.11 module issues addressed to allow it to build again.
- added myself as a chirp maintainer if acceptable.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>chirp</li>
    <li>chirp.dist</li>
    <li>python311Packages.suds</li>
    <li>python311Packages.suds.dist</li>
  </ul>
</details>

**Request**: This is my first time packaging for python with nixpkgs, please let me know if there are best practices or anything I missed on the new package.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/suds/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  fetchPypi,
+  isPyPy,
+  pytestCheckHook,
+  python311Packages,
+}:
+
+buildPythonPackage rec {
+  pname = "suds";
+  version = "1.1.2";
+  format = "setuptools";
```
Comment (wegank) [inline pkgs/development/python-modules/suds/default.nix:13]:
```suggestion
  pyproject = true;
```

`format` has been removed from the [docs](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters), which now refer to `pyproject = true;`.

Also, add `build-system = [ setuptools ];` if necessary.

File: pkgs/development/python-modules/suds/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  isPyPy,
+  pytestCheckHook,
+  python311Packages,
```
Comment (wegank) [inline pkgs/development/python-modules/suds/default.nix:7]:
```suggestion
  six,
```

File: pkgs/development/python-modules/suds/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-HVz6dBFxk7JEpCM/JGxIPZ9BGYtEjF8UqLrRHE9knys=";
+  };
+
+  propagatedBuildInputs = with python311Packages; [
+    six
+  ];
```
Comment (wegank) [inline pkgs/development/python-modules/suds/default.nix:22]:
```suggestion
  dependencies = [ six ];
```

File: pkgs/development/python-modules/suds/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  pythonImportsCheck = [ "suds" ];
+
+  meta = {
+    changelog = "https://github.com/suds-community/suds/blob/v${version}/CHANGELOG.md";
+    description = " Suds is a lightweight SOAP python client for consuming Web Services. A community fork of the jurko fork.";
```
Comment (wegank) [inline pkgs/development/python-modules/suds/default.nix:29]:
```suggestion
    description = "Lightweight SOAP python client for consuming Web Services";
```

https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/meta.chapter.md#description-var-meta-description

File: pkgs/applications/radio/chirp/default.nix [hunk 5]
```diff
@@ -47,7 +48,7 @@ python3.pkgs.buildPythonApplication rec {
     description = "Free, open-source tool for programming your amateur radio";
     homepage = "https://chirp.danplanet.com/";
     license = licenses.gpl3Plus;
-    maintainers = [ maintainers.emantor ];
+    maintainers = with lib.maintainers; [ emantor wrmilling ];
```
Comment (wegank) [inline pkgs/applications/radio/chirp/default.nix:51]:
```suggestion
    maintainers = with maintainers; [ emantor wrmilling ];
```

File: pkgs/development/python-modules/suds/default.nix (hunk 3 above)
Comment (natsukium) [reply pkgs/development/python-modules/suds/default.nix:22]: and `six` should go to `nativeBuildInputs` because it isn't needed in runtime.

**Comment by wrmilling** [comment]
Thank you for the detailed review @wegank.

File: pkgs/development/python-modules/suds/default.nix [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-HVz6dBFxk7JEpCM/JGxIPZ9BGYtEjF8UqLrRHE9knys=";
+  };
+
+  build-system = [ setuptools ];
+
+  nativeBuildInputs = [ six ];
+  nativeCheckInputs = [ pytestCheckHook ];
```
Comment (natsukium) [inline pkgs/development/python-modules/suds/default.nix:24]:
```suggestion
  nativeCheckInputs = [
    pytestCheckHook
    six
  ];
```
Sorry, my mistake. Please add `six` to `nativeCheckInputs`.

File: pkgs/development/python-modules/suds/default.nix [hunk 7]
```diff
@@ -0,0 +1,35 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  isPyPy,
```
Comment (natsukium) [inline pkgs/development/python-modules/suds/default.nix:5]:
```suggestion
```
unused

**Comment by wrmilling** [comment]
Updated @natsukium, thanks as well for the review.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 326171` at a445123f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>chirp</li>
    <li>python311Packages.suds</li>
    <li>python312Packages.suds</li>
  </ul>
</details>

File: pkgs/development/python-modules/suds/default.nix [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-HVz6dBFxk7JEpCM/JGxIPZ9BGYtEjF8UqLrRHE9knys=";
+  };
+
+  build-system = [ setuptools ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    six
```
Comment (dotlambda) [inline pkgs/development/python-modules/suds/default.nix:24]: Are you sure six is needed?

File: pkgs/applications/radio/chirp/default.nix [hunk 9]
```diff
@@ -10,13 +10,13 @@
 
 python3.pkgs.buildPythonApplication rec {
   pname = "chirp";
-  version = "0.4.0-unstable-2024-05-24";
+  version = "0.4.0-unstable-2024-07-05";
```
Comment (dotlambda) [inline pkgs/applications/radio/chirp/default.nix:13]:
```suggestion
  version = "0.4.0-unstable-2024-07-05";
  pyproject = true;
```

File: pkgs/applications/radio/chirp/default.nix [hunk 10]
```diff
@@ -10,13 +10,13 @@
...
 
   src = fetchFromGitHub {
     owner = "kk7ds";
     repo = "chirp";
-    rev = "e17c021ba4fc39eea8a2a1de37ef04a0d1253090";
-    hash = "sha256-YvIRo7g9fxnlf8og5CM2JLf8DeADVkcHdvb4ppS1veE=";
+    rev = "f28814fff0566dcf6f93ac5b7a79d6d594202757";
+    hash = "sha256-SQ0j9DGJc48TsaUaesixJT/pMBm9NGw22qapSokj9r8=";
   };
```
Comment (dotlambda) [inline pkgs/applications/radio/chirp/default.nix:20]:
```suggestion
  };

  build-system = [ setuptools ];
```

File: pkgs/applications/radio/chirp/default.nix (hunk 9 above)
Comment (wrmilling) [reply pkgs/applications/radio/chirp/default.nix:13]: Adding the following to chirp causes the build to fail to find the `suds` package: 

```
pyproject = true;
build-system = with python3.pkgs; [ setuptools ];
```

Any objection to holding off on non-update related changes to the `chirp` package till after this PR?

Comment (wrmilling) [reply pkgs/applications/radio/chirp/default.nix:13]: In the interest of moving the core changes forward I am going to resolve these and plan on a follow-up PR once the chirp package is updated/stable again.

**Review by Emantor — APPROVED** [review]
- [x] builds on NixOS unstable
```
/nix/store/qv7cdiix42i8dhxda7324a01928l3awl-chirp-0.4.0-unstable-2024-07-05
```
- [x] binary works by loading my baofeng config

**Pull request merged**